src/core/messages.ts

```diff
@@ -277,6 +293,130 @@ export interface Messages extends EmitsDiscontinuities {
+  timestamp: Date;
+}
+
+export type MessageReactionListener = (event: MessageReactionSummaryEvent) => void;
+export type MessageRawReactionListener = (event: MessageReactionEvent) => void;
+
+export interface Unsubscribable {
+  unsubscribe: () => void;
+}
+
+interface Reactions {
+  add(message: { serial: string }, refType: ReactionRefType, reaction: string, count?: number): Promise<void>;
+  remove(message: { serial: string }, refType: ReactionRefType, reaction: string): Promise<void>;
+  subscribe(listener: MessageReactionListener): Unsubscribable;
+  subscribeRaw(listener: MessageRawReactionListener): Unsubscribable;
+}
+
+export class DefaultMessageReactions implements Reactions {
+  private _emitter = new EventEmitter<{ reaction: MessageReactionEvent; summary: MessageReactionSummaryEvent }>();
+
+  constructor(
+    private readonly _logger: Logger,
+    private readonly _options: MessageOptions | undefined,
+    private readonly _api: ChatApi,
+    private readonly _roomID: string,
+    private readonly _channel: Ably.RealtimeChannel,
+  ) {
+    void _channel.subscribe(this._processMessageEvent.bind(this));
+    if (this._options?.RawMessageReactions) {
+      void _channel.annotations.subscribe(this._processAnnotationEvent.bind(this));
+    }
+  }
+
+  private _processAnnotationEvent(event: Ably.Annotation) {
+    console.log('ably-chat-js: annotation received:', event);
+  }
+
+  private _processMessageEvent(event: Ably.InboundMessage) {
+    // only process summary events
+    if (event.action !== ChatMessageActions.MessageAnnotationSummary) {
+      return;
+    }
+    // they must have a refSerial
+    if (!event.refSerial) {
+      return;
+    }
+    // they must have a version
+    if (!event.version) {
+      return;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/prefer-optional-chain, @typescript-eslint/no-unsafe-assignment
+    const single: Record<string, SingleReactionSummary> = (event.summary || {})[ReactionRefType.Single] || {};
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/prefer-optional-chain, @typescript-eslint/no-unsafe-assignment
+    const distinct: Record<string, DistinctReactionSummary> = (event.summary || {})[ReactionRefType.Distinct] || {};
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/prefer-optional-chain, @typescript-eslint/no-unsafe-assignment
+    const counter: Record<string, CounterReactionSummary> = (event.summary || {})[ReactionRefType.Counter] || {};
+
+    this._emitter.emit('summary', {
+      type: 'message-reaction-summary',
+      timestamp: new Date(event.timestamp),
+      refSerial: event.refSerial,
+      version: event.version,
+      single: single,
+      distinct: distinct,
+      counter: counter,
+    });
+  }
+
+  add(message: { serial: string }, refType: ReactionRefType, reaction: string, count?: number): Promise<void> {
```

Comment:
I think that having the refType on a per-reaction basis is probably overly-configurable. I can't think of any Chat apps that mix two schemes, and I also don't think a scheme change will come up regularly.

I would suggest that the message reaction mode is controlled by `RoomOptions` and set once at initialisation. If we ever get a case where it needs to be per-message, we can add an optional parameter later.